toolvoice/scripts/inference/cascade_asr/asr_cascade_gemini_confetti.py

```
@@ -0,0 +1,378 @@
+                        if out_trans and getattr(out_trans, "text", None):
+                            assistant_chunks.append(out_trans.text)
+
+                    if getattr(response, "tool_call", None):
+                        for fc in response.tool_call.function_calls or []:
+                            tool_calls.append(
+                                {
+                                    "id": fc.id or "",
+                                    "name": fc.name or "",
+                                    "args_json": json.dumps(fc.args or {}, ensure_ascii=False),
+                                }
+                            )
```

Ruff (0.16.1)

[warning] Use `list.extend` to create a transformed list

(PERF401)